astropy/units/format/vounit.py

```diff
@@ -151,7 +141,10 @@ def _get_unit_name(cls, unit: NamedUnit) -> str:
                     f"In '{unit}': VOUnit can not represent units with the 'd' "
                     "(deci) prefix"
                 )
-        return super()._get_unit_name(unit)
+        name = unit._get_format_name(cls.name)
+        if name not in cls._custom_units:
```

Comment:
style nit, feel free to ignore
```suggestion
        if (name := unit._get_format_name(cls.name)) not in cls._custom_units:
```

Reply:
so, what you mean is that you don't want to use it here because `name` is also used beyond this `if` block ?

Reply:
Yes.